loom/common/utils/utByteArray.cpp

```diff
@@ -82,27 +103,40 @@ void utByteArray::uncompress(int uncompressedSize, int maxBuffer)
  
     stream.next_in = (Bytef *) ( (unsigned char*) getDataPtr());
     stream.avail_in = (uLong) getSize();
-    stream.next_out = (Bytef*) dest.getDataPtr();
+    stream.next_out = (Bytef *) dest.getDataPtr();
     stream.avail_out = sz;
  
-    ok = inflateInit2(&stream, 15 + 32);
-    if (ok != Z_OK)
+    ret = inflateInit2(&stream, 15 + 32); // zlib + gzip autodetection
+    if (ret != Z_OK)
     {
         resize(0);
         return;
     }
- 
-    ok = inflate(&stream, Z_NO_FLUSH);
- 
-    if (ok != Z_STREAM_END)
+    
+    while (true) {
```

Comment:
I think that as weird as it is, it might be the cleanest solution, since while we're getting Z_OK, everything is fine and we just need more space, if we get anything else, the if doesn't run and the loop breaks.

I added documentation saying as much in any case.